addons/l10n_hu_edi/models/account_move.py

```diff
@@ -390,6 +392,16 @@ def _l10n_hu_edi_check_invoices(self):
                 'message': _('The following invoices appear to be earlier in the chain, but have not yet been sent. Please send them first.'),
                 'action_text': _('View invoice(s)'),
             },
+            'invoice_advance_not_paid': {
+                'records': advance_invoices.filtered(
+                    lambda m: (
+                        m.payment_state not in ['in_payment', 'paid', 'partial']
+                        or m.l10n_hu_edi_state in [False, 'rejected', 'cancelled'] and m not in self
```

Comment:
Does `m not in self` make sense ? `advance_invoices` is a `self.filtered()` so every record browsed in `advance_invoice.filtered()` is also a record of self, am I mistaken ?

Reply:
I guess this must have been really confusing, so I've added a comment that explains what the `advance_invoices` are. `advance_invoices` actually contains all the advance invoices that correspond to final invoices in `self`. 

`self.filtered(lambda m: not m._is_downpayment())`: get all invoices in self that are not advance invoices
`.invoice_line_ids._get_downpayment_lines()`: get all the advance invoice lines that correspond to final invoices in this recordset

So, it may contain advance invoices which are in `self` if their final invoice is also in `self`, but it may also contain advance invoices that are not in `self`, provided the final invoice is in `self`.

Reply:
Okay i get it now, thank you for the clarification :+1: